src/utils/decorators/input-validation/__spec__.js

```diff
@@ -393,46 +419,106 @@ describe('InputValidation', () => {
               offsetWidth: 10
             };
             wrapper.find(Component).instance().validationMessage = {
-              className: "",
+              offsetWidth: 10,
+              offsetLeft: 10,
+              offsetHeight: 10,
+              style: {},
+              classList: {
+                remove: removeClassSpy
+              },
+              getBoundingClientRect: () => {
+                return {
+                  top: 0,
+                  left: 0,
+                  width: 300,
+                  height: 30
+                };
+              }
+            };
+
+            wrapper.find(Component).instance().validationIcon._target = {
               offsetWidth: 10,
               offsetLeft: 10,
               offsetHeight: 10,
               style: {},
               getBoundingClientRect: () => {
-                return {};
+                return {
+                  top: 100,
+                  left: 900,
+                  width: 20,
+                  height: 20
+                };
               }
             };
+
             input.simulate('focus');
-            expect(wrapper.find(Component).instance().validationMessage.className).toEqual(' common-input__message--flipped');
+            jest.runOnlyPendingTimers();
+
+            expect(wrapper.find(Component).instance().flipped).toBe(true);
+            expect(wrapper.find(Component).instance().validationMessage.className).toEqual(expect.stringContaining('common-input__message--flipped'));
+            expect(wrapper.find(Component).instance().validationMessage.style.left).toEqual('610px');
+            expect(wrapper.find(Component).instance().validationMessage.style.top).toEqual('50px');
           });
         });
 
         describe('when offscreen', () => {
+          let wrapper;
+          beforeEach(() => {
+            jest.useFakeTimers();
```

Comment:
ditto - could the timers code be moved up the hierarchy?

Reply:
ditto